test/Microsoft.Dnx.Testing.CommonUtils/Exec.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+
+namespace Microsoft.Dnx.Testing
+{
+    public class Exec
+    {
+        public static ExecResult Run(
+            string program,
+            string commandLine,
+            Action<Dictionary<string, string>> envSetup = null,
+            string workingDir = null)
+        {
+            Console.WriteLine($"Running: {program} {commandLine}");
+            var env = new Dictionary<string, string>();
+            envSetup?.Invoke(env);
+
+            var processStartInfo = new ProcessStartInfo()
+            {
+                UseShellExecute = false,
+                WorkingDirectory = workingDir,
+                FileName = program,
+                Arguments = commandLine,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true
+            };
+
+            foreach (var pair in env)
+            {
+#if DNX451
+                processStartInfo.EnvironmentVariables[pair.Key] = pair.Value;
+#else
+                processStartInfo.Environment.Add(pair);
```

Comment:
Will this fail if I already have set one of the environment variables? For example I set `DNX_TRACE` in my console and then I run a test that sets `DNX_TRACE` again. Will this conflict with the inherited variables?